### ISSUE TITLE
  without Ort::Env env(ORT_LOGGING_LEVEL_WARNING, "test") {1, 3, 32, 620} ok {1, 3, 32, 630} crash

### ISSUE THREAD

**Issue by FrisaSZ** [microsoft/onnxruntime#6274]
**Describe the bug**
I have divided the working codes into two functions, first function is class constructor:
```
Ort::Env env(ORT_LOGGING_LEVEL_WARNING, "test");
    Ort::SessionOptions session_options;
    session_options.SetGraphOptimizationLevel(
        GraphOptimizationLevel::ORT_ENABLE_ALL);
    session_ = Ort::Session(env, model_file, session_options);
```
second function is a running function
```
...
  auto output_tensors =
      session_.Run(Ort::RunOptions{nullptr}, input_node_names.data(),
                   &input_tensor, 1, output_node_names.data(), 1);
...
```
when the size of input is {1, 3, 32, 620}, it works, but with an input of size {1, 3, 32, 630}, it crashes.
If I add one line `Ort::Env env(ORT_LOGGING_LEVEL_WARNING, "test")` into the running function, it works too.
the related codes are [here](https://github.com/FrisaSZ/minimum_working_code/blob/a737d573d5d73b8c4a6632aa774fe0f9148fc4c4/onnxruntime/main.cpp#L26-L29)

**Urgency**
none

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04
- ONNX Runtime installed from (source or binary): binary
- ONNX Runtime version: onnxruntime-linux-x64-1.6.0
- Python version:
- Visual Studio version (if applicable):
- GCC/Compiler version (if compiling from source): gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.12)
- CUDA/cuDNN version:
- GPU model and memory:

**To Reproduce**
the full working project is [here](https://github.com/FrisaSZ/minimum_working_code/tree/master/onnxruntime), onnx model is included in the repo
use the following cmds
```
mkdir build
cd build
cmake ..
make
cd ..
./build/main
```
to reproduce

**Expected behavior**

**Screenshots**

**Additional context**
Add any other context about the problem here. If the issue is about a particular model, please share the model details as well to facilitate debugging.

**Comment by hariharans29** [comment]
The lifetime of this env object (https://github.com/FrisaSZ/minimum_working_code/blob/a737d573d5d73b8c4a6632aa774fe0f9148fc4c4/onnxruntime/main.cpp#L13) is only till the ctor is is scope. To use any Ort method, the environment must be valid. Add another member, env_ like session_ to your class and have it initialized. 

 I think when you pass in 630, it is not a vlaid shape and it is trying to use the default logger to log this event and since the env is not valid, it crashes because the default logger is null.

Bottom line: Keep env valid until the last usage of Ort is done.

**Comment by FrisaSZ** [comment]
@hariharans29 Thank you very much.
